test/index.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+
+const assert = require('assert');
+const RedisPool = require('..');
+const { promisify } = require('util');
+
+const MAX_POOLS = 2
+const TEST_OPTIONS = {
+  max: MAX_POOLS,
+  idleTimeoutMillis: 1,
+  reapIntervalMillis: 1,
+  port: 6379
+};
+
+describe('RedisPool', function () {
+  it('RedisPool object exists', function () {
+    assert.ok(RedisPool);
+  });
+
+  it('RedisPool can create new RedisPool objects with default settings', function () {
+    const redisPool = new RedisPool()
+    assert.ok(redisPool)
+  });
+
+  it('RedisPool can create new RedisPool objects with specific settings', function () {
+    const options = Object.assign({ host: '127.0.0.1', port: '6379' }, TEST_OPTIONS)
+    const redisPool = new RedisPool(options)
+    assert.ok(redisPool)
+  });
+
+  it('pool has proper size, available and pending', async function () {
+    const DATABASE = 0
+
+    const options = Object.assign(TEST_OPTIONS)
+    const redisPool = new RedisPool(options)
+
+    const client1 = await redisPool.acquire(DATABASE)
+    const client2 = await redisPool.acquire(DATABASE)
+
+    let pool = redisPool.pools[DATABASE]
+
+    assert.equal(pool.size, 2)
+    assert.equal(pool.available, 0)
+    assert.equal(pool.pending, 0)
+
+    await redisPool.release(0, client1); // needed to exit tests
+    await redisPool.release(0, client2); // needed to exit tests
+  });
+
+  it('new command only works after adding it to Redis', async function () {
+    const NEW_COMMAND = 'fakeCommand'
+
+    let redisPool = new RedisPool(TEST_OPTIONS)
+    let client = await redisPool.acquire(0)
+    assert.strictEqual(client[NEW_COMMAND], undefined);
+    await redisPool.release(0, client);
+
+    const options = Object.assign(
+      TEST_OPTIONS,
+      { commands: ['fakeCommand'] }
+    );
+    redisPool = new RedisPool(options);
+
+    client = await redisPool.acquire(0)
+    const fakeCommand = promisify(client[NEW_COMMAND]).bind(client);
+
+    const response = await fakeCommand('key').catch(async (error) => {
+      assert.equal(error.name, 'ReplyError');
+      assert.ok(error.message.startsWith("ERR unknown command"));
+      assert.ok(error.message.includes('fakeCommand'));
+    })
+
+    assert.ok(response === undefined)
+
+    await redisPool.release(0, client); // needed to exit tests
+  });
+
+  it('pool object has an acquire function', function () {
+    const redisPool = new RedisPool(TEST_OPTIONS)
+    assert.ok(typeof redisPool.acquire === 'function');
+  });
+
+  it('calling aquire returns a redis client object that can get/set', async function () {
+    const redisPool = new RedisPool(TEST_OPTIONS)
+    const client = await redisPool.acquire(0)
+
+    const set = promisify(client.set).bind(client);
```

Comment:
Never used `.bind(client)` after promisifying, shouldn't be: 

```suggestion
    const set = promisify(client.set.bind(client));
```